source/Worlds/MazeWorld.hpp

```diff
@@ -19,25 +20,51 @@ namespace cse491 {
     size_t floor_id; ///< Easy access to floor CellType ID.
     size_t wall_id;  ///< Easy access to wall CellType ID.
 
+    size_t spike_id;  ///< Easy access to spike CellType ID.
+    size_t tar_id; ///< Easy access to tar CellTypeID
+
+    size_t grass_id;
+    size_t dirt_id;
+
+    unsigned int seed; ///< Seed used for generator
+    std::mt19937 randomGenerator;
+
+
     /// Provide the agent with movement actions.
     void ConfigAgent(AgentBase & agent) override {
       agent.AddAction("up", MOVE_UP);
       agent.AddAction("down", MOVE_DOWN);
       agent.AddAction("left", MOVE_LEFT);
       agent.AddAction("right", MOVE_RIGHT);
+      agent.SetProperty("tar_property", 5.0); //if it is set to 5.0, agent is free to move. if it is set to 6.0, agent is stuck
     }
 
   public:
-    MazeWorld() {
+    /**
+     * Initializes world with cell types and random generator
+     * @param seed Seed used for RNG. Default is current time
+     */
+    explicit MazeWorld(unsigned int seed = time(nullptr)) : seed(seed) {
       floor_id = AddCellType("floor", "Floor that you can easily walk over.", ' ');
       wall_id = AddCellType("wall", "Impenetrable wall that you must find a way around.", '#');
+
+      spike_id = AddCellType("spike", "Dangerous spike that resets the game.", 'X');
+
+      tar_id = AddCellType("tar", "Slow tile that makes you take two steps to get through it", 'O');
+
+      grass_id = AddCellType("grass", "Grass you can walk on.", 'M');
+      dirt_id  = AddCellType("dirt", "Dirt you can walk on.", '~');
+
       main_grid.Read("../assets/grids/default_maze.grid", type_options);
+
+      randomGenerator.seed(seed);
     }
     ~MazeWorld() = default;
 
     /// Allow the agents to move around the maze.
     int DoAction(AgentBase & agent, size_t action_id) override {
       // Determine where the agent is trying to move.
+      GridPosition currentPosition = agent.GetPosition();
```

Comment:
For consistency of style, I would call this variable `cur_position` or something along those lines.